packages/two_dimensional_scrollables/lib/src/tree_view/tree.dart

```diff
@@ -0,0 +1,1056 @@
+  /// are going to manually maintain their [KeepAlive] state. It may also be
+  /// more efficient to disable this feature if it is known ahead of time that
+  /// none of the children will ever try to keep themselves alive.
+  ///
+  /// Defaults to true.
+  final bool addAutomaticKeepAlives;
+
+  /// Whether to wrap each row in a [RepaintBoundary].
+  ///
+  /// Typically, children in a scrolling container are wrapped in repaint
+  /// boundaries so that they do not need to be repainted as the list scrolls.
+  /// If the children are easy to repaint (e.g., solid color blocks or a short
+  /// snippet of text), it might be more efficient to not add a repaint boundary
+  /// and instead always repaint the children during scrolling.
+  ///
+  /// Defaults to true.
+  final bool addRepaintBoundaries;
+
+  /// Used to override the toggle animation's curve and duration.
+  ///
+  /// If [AnimationStyle.duration] is provided, it will be used to override
+  /// the [TreeView.defaultAnimationDuration], which is 150
+  /// milliseconds.
+  ///
+  /// If [AnimationStyle.curve] is provided, it will be used to override
+  /// the [TreeView.defaultAnimationCurve], which is [Curves.linear].
+  ///
+  /// To disable the tree animation, use [AnimationStyle.noAnimation].
+  final AnimationStyle? animationStyle;
+
+  /// A default of [Curves.linear], which is used in the tree's expanding and
+  /// collapsing node animation.
+  static const Curve defaultAnimationCurve = Curves.linear;
+
+  /// A default [Duration] of 150 milliseconds, which is used in the tree's
+  /// expanding and collapsing node animation.
+  static const Duration defaultAnimationDuration = Duration(milliseconds: 150);
+
+  /// The order in which [TreeViewNode]s are visited.
+  ///
+  /// This value will influence [TreeViewport.mainAxis] so nodes are traversed
+  /// properly when they are converted to a [TreeVicinity] in the context of the
+  /// active nodes of the tree.
+  ///
+  /// Defaults to [TreeViewTraversalOrder.depthFirst].
+  final TreeViewTraversalOrder traversalOrder;
+
+  /// The number of pixels children will be offset by in the cross axis based on
+  /// their [TreeViewNode.depth].
+  ///
+  /// By default, the indentation is handled by [RenderTreeViewport]. Child
+  /// nodes are offset by the indentation specified by
+  /// [TreeViewIndentationType.value] in the cross axis of the viewport. This
+  /// means the space allotted to the indentation will not be part of the space
+  /// made available to the Widget returned by [TreeView.treeNodeBuilder].
+  ///
+  /// Alternatively, the indentation can be implemented in
+  /// [TreeView.treeNodeBuilder], with the depth of the given tree row accessed
+  /// by [TreeViewNode.depth]. This allows for more customization in building
+  /// tree rows, such as filling the indented area with decorations or ink
+  /// effects.
+  final TreeViewIndentationType indentation;
```

Comment:
I wanted to make this its own type, because I anticipate future feature requests around managing tree indentation. I also liked the semantics it provided the user over a plain number value.